src/main/java/pl/psnc/dei/model/Transcription.java

```diff
@@ -1,41 +1,60 @@
 package pl.psnc.dei.model;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.AccessLevel;
-import lombok.Data;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.Setter;
+import lombok.*;
 import org.apache.jena.atlas.json.JsonObject;
 
 import javax.persistence.*;
+import java.io.Serializable;
 
 @Entity
-@Data
-@NoArgsConstructor
 public class Transcription {
 
-	@Id
-	@GeneratedValue
-	@Getter(AccessLevel.NONE)
-	@Setter(AccessLevel.NONE)
-	private long id;
-
-	private String tp_id;
-
+	@EmbeddedId
+	private final TranscriptionPK key;
 	@ManyToOne(fetch = FetchType.LAZY)
+	@Getter
+	@Setter
 	private Record record;
-
 	@Transient
+	@Getter
+	@Setter
 	private JsonObject transcriptionContent;
 
+	public Transcription(String tp_id, Record record, String annotationId) {
+		this.key = new TranscriptionPK(tp_id, annotationId);
+		this.record = record;
+	}
+
+	public Transcription() {
+		this.key = new TranscriptionPK();
+	}
+
+	public String getTp_id() {
+		return this.key.getTp_id();
+	}
+
+	public void setTp_id(String tp_id) {
+		this.key.setTp_id(tp_id);
+	}
+
 	@JsonProperty("EuropeanaAnnotationId")
-	private String annotationId;
+	public String getAnnotationId() {
+		return this.key.annotationId;
+	}
 
+	public void setAnnotationId(String annotationId) {
+		this.key.setAnnotationId(annotationId);
+	}
 
-	public Transcription(String tp_id, Record record, String annotationId) {
-		this.tp_id = tp_id;
-		this.record = record;
-		this.annotationId = annotationId;
+	@Data
+	@Embeddable
+	@AllArgsConstructor
+	@NoArgsConstructor
+	public static class TranscriptionPK implements Serializable {
+		@Column(length = 120)
+		private String tp_id;
+		@Column(length = 120)
+		private String annotationId;
```

Comment:
Does this complex key accept situations when annotationId is null? This is normal situation because annotation id comes from Europeana after successfull transcription ingestion and that may happen later...